vagrant/provisioning/roles/solr/tasks/main.yml

```diff
@@ -97,4 +97,84 @@
 
 - name: Expose required ports for solr
   include_tasks: firewall.yml
-  when: enable_firewall is undefined or enable_firewall
\ No newline at end of file
+  when: enable_firewall is undefined or enable_firewall
+
+- name: find prometheus-exporter log4j jars that are vulnerable
+  find:
+    paths: "{{ root_folder }}/app/solr/contrib/prometheus-exporter/lib"
+    patterns: "log4j*"
```

Comment:
This change is not idempotent, since you are removing files `log4j*` and replacing them with files that also match the pattern `log4j*`.  As far as I can tell every time you run this play, the log4j libraries will be removed and replaced, even with the same version every time.